packages/pluggableWidgets/accordion-web/src/utils/AccordionGroupStateReducer.ts

```diff
@@ -0,0 +1,41 @@
+import { AccGroup } from "../components/AccordionGroup";
+
+export type AccordionGroupsReducerAction =
+    | { type: "sync"; groups: AccGroup[] }
+    | { type: "expand" | "collapse"; group: AccGroup };
+
+export function getAccordionGroupsReducer(
+    expand: "single" | "multiple" | "all"
+): (state: AccGroup[], action: AccordionGroupsReducerAction) => AccGroup[] {
+    return (state: AccGroup[], action: AccordionGroupsReducerAction): AccGroup[] => {
```

Comment:
Nit: For future reference: a switch case is often times more readable since it forces you to split the cases. Now a few cases are intertwined which makes it slightly hard to follow.

Reply:
I agree. I will try that next time :)